File: solutions-by-language/java/1935_maximum_number_of_words_you_can_type.java
public class Solution {
    public static int canBeTypedWords(String text, String brokenLetters) {
        boolean[] broken = new boolean[26];
        for (char ch : brokenLetters.toCharArray()) {
            broken[ch - 'a'] = true;
        }

        int count = 0;
        boolean canType = true;

        for (char ch : text.toCharArray()) {
            if (ch == ' ') {
                if (canType) count++;
                canType = true;
            } else if (broken[ch - 'a']) {
                canType = false;
            }
        }

        if (canType) count++; 

        return count;
    }
}

File: solutions-by-language/cpp/1935_maximum_number_of_words_you_can_type.cpp
class Solution {
public:
    int canBeTypedWords(string text, string brokenLetters) {
        std::stringstream iss(text);
        std::string word;
        int count = 0;
        while(iss >> word){
            if(!isBroken(word,brokenLetters)){
                count++;
            }
        }
        return count;
    }

    bool isBroken(string word, string bl){
        
        for(char ch : bl){
            if(word.find(ch) != std::string::npos ){
                return true;
            }
        }
        
        return false;
    }
};